expression/collation.go

```diff
@@ -68,17 +68,7 @@ func (c *collationInfo) SetCharsetAndCollation(chs, coll string) {
 	c.charset, c.collation = chs, coll
 }
 
-func (c *collationInfo) CharsetAndCollation(ctx sessionctx.Context) (string, string) {
-	if c.charset != "" || c.collation != "" {
-		return c.charset, c.collation
-	}
-
-	if ctx != nil && ctx.GetSessionVars() != nil {
```

Comment:
Why they are removed? Are you sure it is OK to make the change?

Reply:
As I said in 'Problem summary':

When we call CharsetAndCollation, c.charset and c.collation can't be nil. So the rest of code can't be reached.